app/digests/tests/test_digest_api_updates.py

```diff
@@ -0,0 +1,158 @@
+    assert response.status_code == 200
+    assert response.data['subjects'][0]['name'] == new_name
+
+
+@pytest.mark.django_db
+@pytest.mark.freeze_time('2018-01-01 00:00:00')
+def test_can_update_digest_image_via_patch(digest: Digest,
+                                           digest_image_json: Dict,
+                                           rest_client: APIClient):
+    new_image = deepcopy(digest_image_json)
+    new_alt = 'some alt'
+    new_image['thumbnail']['alt'] = new_alt
+    response = rest_client.patch(f'{DIGESTS_URL}/{digest.id}',
+                                 data=json.dumps({'image': new_image}),
+                                 content_type=settings.DIGEST_CONTENT_TYPE)
+    assert response.status_code == 200
+    assert response.data['image']['thumbnail']['alt'] == new_alt
+
+
+@pytest.mark.parametrize('key, value', [
+    ('image', {}),
+    ('subjects', [{"foo": "bar"}]),
+    ('content', []),
+    ('relatedContent', {}),
+])
+@pytest.mark.django_db
+@pytest.mark.freeze_time('2018-01-01 00:00:00')
+def test_will_reject_invalid_json_field_updates(key: str,
+                                                value: str,
+                                                digest: Digest,
+                                                digest_json: Dict,
+                                                rest_client: APIClient):
+    response = rest_client.patch(f'{DIGESTS_URL}/{digest.id}',
+                                 data=json.dumps({key: value}),
+                                 content_type=settings.DIGEST_CONTENT_TYPE)
+    assert response.status_code == 400
+
+
+@pytest.mark.django_db
+@pytest.mark.freeze_time('2018-01-01 00:00:00')
+def test_can_update_digest_via_put(digest: Digest,
+                                   digest_json: Dict,
+                                   rest_client: APIClient):
+    new_data = deepcopy(digest_json)
+    new_data['title'] = 'New Title'
+    new_data['stage'] = 'published'
+
+    response = rest_client.put(f'{DIGESTS_URL}/{digest.id}',
+                               data=json.dumps(new_data),
+                               content_type=settings.DIGEST_CONTENT_TYPE)
+    assert response.status_code == 200
+    assert response.data['title'] == 'New Title'
+    assert response.data['stage'] == 'published'
+    assert response.data['updated'] == '2018-01-01T00:00:00Z'
+
+
+@pytest.mark.django_db
+@pytest.mark.freeze_time('2018-01-01 00:00:00')
+def test_reject_invalid_data_via_put(digest: Digest,
+                                     digest_json: Dict,
+                                     rest_client: APIClient):
+    new_data = deepcopy(digest_json)
+    new_data['content'] = {}
+    new_data['image'] = {}
+
+    response = rest_client.put(f'{DIGESTS_URL}/{digest.id}',
+                               data=json.dumps(new_data),
+                               content_type=settings.DIGEST_CONTENT_TYPE)
+    assert response.status_code == 400
```

Comment:
good test coverage. `POST`, `PUT` and `PATCH` are getting subtle enough that they should be included in https://github.com/elifesciences/api-raml/blob/develop/src/api.raml#L807-L863 to document the contract